pkg/abstract/slices.go

```diff
@@ -25,3 +25,23 @@ func SliceDelete[Elt any](slice *[]Elt, index int) {
 	}
 	*slice = (*slice)[:lastIndex]
 }
+
+// SliceCopy copies a given slice into new storage
+func SliceCopy[Elt any](input []Elt) []Elt {
+	if input == nil {
+		return nil
+	}
+	return append(make([]Elt, 0, len(input)), input...)
+}
+
+func SliceEqual[Elt comparable](slice1, slice2 []Elt) bool {
```

Comment:
Why do we need the Slice prefix?  Isn't IsEqual  while the parameters are slices good enough ?

Reply:
We have both "SliceEqual" and "PrimitiveMapEqual".